openvino/itt/include/openvino/itt.hpp

```diff
@@ -0,0 +1,173 @@
+// limitations under the License.
+//*****************************************************************************
+
+/**
+ * @brief Defines API to trace using Intel ITT.
+ * @file itt.hpp
+ */
+
+#pragma once
+#include "function_name.hpp"
+#include "macro_overload.hpp"
+#include <string>
+
+namespace openvino
+{
+    namespace itt
+    {
+        /**
+         * @typedef domain_t
+         * @ingroup ie_dev_profiling
+         * @brief A domain type which enables tagging trace data for different modules or libraries in a program.
+         */
+        typedef struct domain_ {} *domain_t;
+
+        /**
+         * @typedef handle_t
+         * @ingroup ie_dev_profiling
+         * @brief Annotation handle for section of code which would be named at runtime.
+         */
+        typedef struct handle_ {} *handle_t;
+
+/**
+ * @cond
+ */
+        namespace internal
+        {
+            domain_t domain(char const* name);
+            handle_t handle(char const* name);
+            void taskBegin(domain_t d, handle_t t);
+            void taskEnd(domain_t d);
+            void threadName(const char* name);
+        }
+/**
+ * @endcond
+ */
+
+        /**
+         * @fn void threadName(const char* name)
+         * @ingroup ie_dev_profiling
+         * @brief Set thread name using a char string.
+         * @param name [in] The thread name
+         */
+        inline void threadName(const char* name)
+        {
+            internal::threadName(name);
+        }
+
+        inline void threadName(const std::string &name)
+        {
+            internal::threadName(name.c_str());
+        }
+
+        inline handle_t handle(char const *name)
+        {
+            return internal::handle(name);
+        }
+
+        inline handle_t handle(const std::string &name)
+        {
+            return internal::handle(name.c_str());
+        }
+
+        /**
+         * @fn handle_t handle(char const *name)
+         * @ingroup ie_dev_profiling
+         * @brief Create annotation handle with a given name.
+         * @details If template function is instantiated with a tag, the handle is created as a singleton.
+         * @param name [in] The annotation name
+         */
+        template <typename Tag>
+        handle_t handle(char const *name)
+        {
+            static auto h = internal::handle(name);
+            return h;
+        }
+
+        template <typename Tag>
+        handle_t handle(const std::string &name)
+        {
+            return handle<Tag>(name.c_str());
+        }
+
+        template <typename Tag>
+        handle_t handle(handle_t h)
+        {
+            return h;
+        }
+
+        /**
+         * @class ScopedTask
+         * @ingroup ie_dev_profiling
+         * @brief Used to annotate section of code which would be named at runtime
+         */
+        template <domain_t(*domain)()>
+        struct ScopedTask
+        {
+        /**
+        * @brief Construct ScopedTask with defined annotation handle
```

Comment:
1. misaligned
2. Parameter is not documented

Reply:
ok